### ISSUE TITLE
Feature request: "Slice out segment" tool

### ISSUE THREAD

**Issue by twMat** [Jermolene/TiddlyWiki5#2269]
This is a proposal for a generally useful tool but it also enables "different route" from the [text slicer](http://tiddlywiki.com/editions/text-slicer/index.html) edition, to insert complex documents into a TW.

**Proposal**
A tool that lets a user (easily) select a text segment in a longer text or code and *slice it out* so it becomes a separate tiddler. Ideally the tool is usable from both view mode and edit mode.

The original location might thereafter show a transclusion of the new tiddler or a link to it or similar. That is a topic for another issue.

**The problem** is at least two-dimensional:

* It is unfortunate, if not even patronizing, to *force* users to chunk up longer docs in order for TW to be useful.
* What constitutes a "smallest meaningful chunk" depends on context. Forcing the user to chunk up even forces him to chunk it up into pieces where he does not know if they are meaningful at all! 

In some situations, a long complex document can be all that is needed - but it should be super simple to slice out arbitrary segments in this into separate tiddlers **on a need basis**. We simply lack a good *"cut out and tiddlify this section"*-tool for this.

**Side note**
It can even be argued if the core or tw.com really sticks to tiddler philosophy; For example, the typical stylesheet is definitely one complex single tiddler. And the edit view of most documentation consists of several "chunks" that do different things (a text chunk, a filter chunk, ...)
<br>

**P.S** I would prefer to title this thread *Feature request: "__Tiddlify__ segment" tool*, because to me *tiddlify* sounds like it means "turning something into a tiddler". Is this appropriate or does *tiddlify* really mean something else?

**Comment by Jermolene** [comment]
> It is unfortunate, if not even patronizing

No. As I've said before, handling anchors within long tiddlers isn't blocked by my philosophical position. I just don't see a way to implement them alongside the existing permalink handling. It's a technical problem, and it's better to speak about it as such.

**Comment by Jermolene** [comment]
I think you've puffed up a simple request into something much less useful. Why not just say this:

"I'd like a means to select text within a tiddler and choose a button to extract that text into a separate tiddler. It would be useful to help me gradually cut longer tiddlers up into smaller chunks by extracting the parts I want to keep. Ideally the tool would be usable in both edit mode and view mode."

That's what I mean about the need for tickets to be understandable and actionable. As the developer, otherwise I have to wade through a whole lot of opinions and assertions that are entirely distracting from the technical proposal and discussion.

For example, here you are making several misapprehensions:

> It can even be argued if the core or tw.com really sticks to tiddler philosophy

It's not a law that we have to keep tiddlers short, it's advice that that is the best way to make them reusable. TiddlyWiki supports long tiddlers and there are situations where they are the right thing to use.

> What constitutes a "smallest meaningful chunk" depends on context.

Of course it does! That's the point, that's why there is no character limit.

> Forcing the user to chunk up even forces him to chunk it up into pieces where he does not know if they are meaningful at all!

Nobody is forced to chunk up anything. We've got an outstanding issue that there's no support for internal anchor links, but that's not the same thing.

> In some situations, a long complex document can be all that is needed - but it should be super simple to slice out arbitrary segments in this into separate tiddlers on a need basis. We simply lack a good "cut out and tiddlify this section"-tool for this.

Here you're just restating your original suggestion, but in a shouty tone, and not adding anything new to the readers understanding.

> I would prefer to title this thread Feature request: "Tiddlify segment" tool, because to me tiddlify sounds like it means "turning something into a tiddler". Is this appropriate or does tiddlify really mean something else?

Creating new words in issue titles doesn't seem a great idea; it means that anyone reading the title won't understand what the ticket means unless they take the trouble to learn the vocabulary you are suggesting.

**Comment by twMat** [comment]
Thanks for input!

>> It is unfortunate, if not even patronizing

>No. As I've said before, handling anchors within long tiddlers isn't blocked by my philosophical position.

I fear other issue threads are overshadowing the inerpretation of my OP. //This// proposal is not about anchors at all!

**Comment by Jermolene** [comment]
@twMat your fuller quote:

> It is unfortunate, if not even patronizing, to force users to chunk up longer docs in order for TW to be useful.

We don't force users to chunk longer docs. Long tiddlers work perfectly well within TW, from the feedback I've seen, apart from the lack anchor links. So, what do you mean by those statements? How do we force users? Why is it unfortunate? What is patronizing about it?

The answers to those questions are interesting but you'll note nothing to do with the proposal that you're trying to make.

**Comment by twMat** [comment]
Ok, first, my apologies for clearly not communicating my proposal in a good way. It is intended to be constructive but, well, it didn't come out quite right.

What I mean with "It is unfortunate, if not even patronizing" is that the system(!) - as far as I can tell - does force the user to chunk things up in to smaller segments *for it to be (optimally) useful*. For instance

* The tiddler toolbar is stuck at top; to click edit or to save a tiddler one must scroll up and therefore will loose the point one is at. 
* it is not possible to enter a text into edit mode at a particular point.
* ...and, of course, part of the problem in working with longer tiddlers comes from the lack of the very feature I'm requesting.
 
It is "unfortunate/forcing/patronizing" in that in order to use TW in an efficient way, the user must adapt his content if he wants to use the tools effectively.

One could argue that this is inevitable - any software, or even technical solution, brings with it restrictions and "conditions" for how it can be used. I just think that TW has potential to let longer tiddlers be more easily used in the natural workflow.

Again, I apologize for the way my OP came out and I will reformulate it. I do however believe it is valueable with some background for why a proposal is stated - or would it be better to put such proposals in the tw-dev forum?

**Comment by twMat** [comment]
I'm closing this thread while working on a re-formulation.

**Comment by Jermolene** [comment]
> It is "unfortunate/forcing/patronizing" in that to use TW in an efficient way, the user must adapt his content to the system if he wants to use the tools effectively.

But no tool can do everything. If you want to write a book in Excel you can, but you'd have to "adapt your content to the system". Every tool has a sweetspot. The good ones operate correctly outside their immediate comfort area.

And that doesn't square with the word "force" that you used above. A question of adapting to a tool to get the best from it is different from being "forced" to do something.

>  I just think that TW has potential to let longer tiddlers be more easily used in the natural workflow.

But you're arguing as if there are people who don't want to improve the usability of long tiddlers. Why is that? Plus, what are the actual problems with long tiddlers?

**Comment by twMat** [comment]
>A question of adapting to a tool to get the best from it is different from being "forced" to do something.

Not sure what else to call a situation where "if you don't do this, then this will not be optimal". An "offer"?

>But you're arguing as if there are people who don't want to improve the usability of long tiddlers. Why is that? 

I think it is more a matter of being able to choose the point in time - or several points in time actually - for when "to improve the usability of the long tiddler". The contexts where the to-be tiddlers are needed may not exist yet. So for an indefinite time one should ideally be able to use the longer tiddler with minimal loss of "optimal TW power".

>Plus, what are the actual problems with long tiddlers?

I believe I gave some examples above, but maybe I misunderstand you question?

**Comment by twMat** [comment]
Ok, #2270 posted instead.

I don't know if this thread here should be deleted or just stay closed for possible reference. Either way is fine with me.
